docs/example-react-router.md

```diff
@@ -32,11 +32,72 @@ function App() {
     </div>
   )
 }
+```
 
-// Ok, so here's what your tests might look like
+```jsx
+// app.test.js
+import { Router } from 'react-router-dom'
+import { createMemoryHistory } from 'history'
 
-// this is a handy function that I would utilize for any component
-// that relies on the router being in context
+test('full app rendering/navigating', () => {
+  const history = createMemoryHistory()
+  const { container, getByText } = render(
+    <Router history={history}>
+      <App />
+    </Router>
+  )
+  // verify page content for expected route
+  // often you'd use a data-testid or role query, but this is also possible
+  expect(container.innerHTML).toMatch('You are home')
+
+  fireEvent.click(getByText(/about/i))
+
+  // check that the content changed to the new page
+  expect(container.innerHTML).toMatch('You are on the about page')
+})
+
+test('landing on a bad page shows 404 page', () => {
+  const history = createMemoryHistory()
+  history.push('/some/bad/route')
+  const { getByRole } = render(
+    <Router>
+      <App />
+    </Router>
+  )
+  expect(getByRole('heading')).toHaveTextContent('404 Not Found')
+})
+
+test('rendering a component that uses withRouter', () => {
+  const route = '/some-route'
+  window.history.pushState({}, '', route)
+  const { getByTestId } = render(
+    <Router>
+      <LocationDisplay />
+    </Router>
+  )
+  expect(getByTestId('location-display').textContent).toBe(route)
+})
+```
+
+## Reducing boilerplate
+
+1. You can use the `wrapper` option to wrap a `MemoryRouter` around the component you want to render (`MemoryRouter` works when you don't need access to the history object itself in the test, but just need the components to be able to render and navigate).
+
+```jsx
+import { MemoryRouter } from 'react-router-dom'
+
+test('full app rendering/navigating', () => {
+  const { container, getByText } = render(<App />, {wrapper: MemoryRouter})
+  // verify page content for expected route
+  expect(getByRole('heading')).toMatch('Home')
+})
+```
+
+2. If you find yourself adding Router components to your tests a lot, you may want to create
+a helper function that wraps around `render`. 
+
+```jsx
+// test utils file
 function renderWithRouter(
```

Comment:
Let's use the wrapper for this as well so people get rerender working by default.